### PR TITLE
Wazuh-db socket timeout validation

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#1309|

## Description

This PR adds new tests to: 

- Verify a 0 bytes response is received due to socket close by timeout.
- Cover the new wazuh-db commands to get packages and hotfixes. 

## Tests

- [X] Proven that tests **pass** when they have to pass.
- [X] Proven that tests **fail** when they have to fail.

- [X] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [X] Python codebase is documented following the Google Style for Python docstrings.
- [ ] The test is documented in wazuh-qa/docs.
- [ ] `provision_documentation.sh` generate the docs without errors.

### CODE REVIEW

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 1]
```diff
@@ -76,6 +77,33 @@ def pre_insert_agents():
         assert data[0] == 'ok', f'Unable to update agent {id}'
 
 
+@pytest.fixture(scope="function")
+def pre_set_sync_info():
+    command = f'agent 000 sql UPDATE sync_info SET last_completion = \
+                (SELECT last_attempt from sync_info where component = "syscollector-packages") \
+                where component = "syscollector-packages"'
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:84]:
```suggestion
    command = 'agent 000 sql UPDATE sync_info SET last_completion = (SELECT last_attempt from sync_info ' \
              'where component = "syscollector-packages") where component = "syscollector-packages" '
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 2]
```diff
@@ -76,6 +77,33 @@ def pre_insert_agents():
         assert data[0] == 'ok', f'Unable to update agent {id}'
 
 
+@pytest.fixture(scope="function")
+def pre_set_sync_info():
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:81]: Docstrings are missing.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 3]
```diff
@@ -76,6 +77,33 @@ def pre_insert_agents():
...
+                (SELECT last_attempt from sync_info where component = "syscollector-packages") \
+                where component = "syscollector-packages"'
+    receiver_sockets[0].send(command, size=True)
+    response = receiver_sockets[0].receive(size=True).decode()
+    data = response.split(" ", 1)
+    assert data[0] == 'ok', 'Unable to set sync_info table'
+
+
+@pytest.fixture(scope="function")
+def pre_insert_packages():
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:92]: Docstrings are missing here.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 4]
```diff
@@ -76,6 +77,33 @@ def pre_insert_agents():
...
+    PACKAGES_NUMBER = 2000
+    for pkg_n in range(0, PACKAGES_NUMBER):
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:94]:
```suggestion
    for pkg_n in range(PACKAGES_NUMBER):
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 5]
```diff
@@ -137,3 +165,26 @@ def send_chunk_command(command):
...
     # Check disconnect-agents chunk limit
     send_chunk_command('global disconnect-agents 0 {} syncreq'.format(str(int(time.time()) + 1)))
+
+
+def test_wazuh_db_timeout(configure_sockets_environment,
+                          connect_to_sockets_module,
+                          pre_insert_packages,
+                          pre_set_sync_info):
+
+    """Check that efectively the socket is closed after timeout is reached"""
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:175]:
```suggestion
def test_wazuh_db_timeout(configure_sockets_environment, connect_to_sockets_module, 
                          pre_insert_packages, pre_set_sync_info):
    """Check that efectively the socket is closed after timeout is reached"""
```

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 6]
```diff
@@ -137,3 +165,26 @@ def send_chunk_command(command):
...
+
+    def send_row_by_row_command(command):
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:177]: Mmm... Here I would prefer to remove this function and use it directly.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 7]
```diff
@@ -137,3 +165,26 @@ def send_chunk_command(command):
...
+        receiver_sockets[0].send(command, size=True)
+        loop = True
+        time.sleep(5)
+        while loop:
+            response = receiver_sockets[0].receive(size=True).decode()
+            status = response.split(" ", 1)[0]
+            if status != 'due':
+                loop = False
+                assert response == 'Socket closed', 'ok'
+                break
```
Comment (BraulioV) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:187]:
```suggestion
        time.sleep(5)
        while True:
            response = receiver_sockets[0].receive(size=True).decode()
            status = response.split(" ", 1)[0]
            if status != 'due':
                assert response == 'Socket closed', 'ok'
                break
```
With `break` you actually exits the loop, so you can remove the variable.

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 6 above)
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Done

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 7 above)
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Totally! tanks

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 8]
```diff
@@ -137,3 +167,22 @@ def send_chunk_command(command):
     send_chunk_command('global get-agents-by-connection-status 0 active')
     # Check disconnect-agents chunk limit
     send_chunk_command('global disconnect-agents 0 {} syncreq'.format(str(int(time.time()) + 1)))
+
+
+def test_wazuh_db_timeout(configure_sockets_environment, connect_to_sockets_module,
+                          pre_insert_packages, pre_set_sync_info):
+    """Check that efectively the socket is closed after timeout is reached"""
```
Comment (palaciosjeremias) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:174]: This should be renamed like
_"Check that effectively the socket is closed after timeout is reached"_

**Review by palaciosjeremias — APPROVED** [review]

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 9]
```diff
@@ -172,62 +172,129 @@
...
+    output: ["ok []"]
     stage: "agent sys_osinfo cleaning sys_osinfo table"
   -
     input: 'agent 000 osinfo get'
-    output: "ok []"
+    output: ["ok []"]
     stage: "agent sys_osinfo checking table is empty"
+-
+  name: "Agents' Packages/Hotfixes tables: sys_programs and sys_hotfixes"
+  description: "It checks the commands get packages, set package triage and get hotfixes"
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 10]
```diff
@@ -172,62 +172,129 @@
...
+    output: ['ok']
+    stage: "agent sys_programs adding dummy package"
+  -
+    input: 'agent 003 sql select count(*) from sys_programs'
+    output: ['ok [{"count(*)":2}]']
+    stage: "agent sys_programs count packages added"
+  -
+    input: 'agent 003 package get'
+    output: ['ok {"status":"NOT_SYNCED"}']
+    stage: "agent sys_programs getting packages attempt"
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 11]
```diff
@@ -172,62 +172,129 @@
...
+    input: 'agent 003 hotfix save 0|0|KB2980295|legacy'
+    output: ['ok']
+    stage: "agent sys_hotfixes adding dummy hotfix"
+  -
+    input: 'agent 003 sql SELECT count(*) FROM sys_hotfixes'
+    output: ['ok [{"count(*)":3}]']
+    stage: "agent sys_hotfixes count hotfixes added"
+  -
+    input: 'agent 003 hotfix get'
+    output: ['ok {"status":"NOT_SYNCED"}']
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml:272]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 12]
```diff
@@ -172,62 +172,129 @@
...
+    stage: "agent sys_hotfixes getting hotfixes attempt"
+  -
+    input: 'agent 003 sql UPDATE sync_info SET last_attempt = 1, last_completion = 1 where component = "syscollector-hotfixes"'
+    output: ['ok []']
+    stage: "agent sync_info last_attempt and last_completion fields match for hotfixes."
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 13]
```diff
@@ -172,62 +172,129 @@
...
+  -
+    input: 'agent 003 hotfix get'
+    output: ['due {"hotfix":"KB2980293"}','due {"hotfix":"KB2980294"}','due {"hotfix":"KB2980295"}','ok {"status":"SUCCESS"}']
+    stage: "agent sys_hotfixes getting hotfixes"
+  -
+    input: 'agent 003 sql UPDATE sync_info SET last_attempt = 1, last_completion = 1 where component = "syscollector-packages"'
+    output: ['ok []']
+    stage: "agent sync_info last_attempt and last_completion fields match for packages."
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 14]
```diff
@@ -172,62 +172,129 @@
...
+  -
+    input: 'agent 003 sql UPDATE sys_programs SET triaged = 1  WHERE name = "test_rpm_pkg"'
+    output: ['ok []']
+    stage: "agent sys_programs set package as triaged"
+  -
+    input: 'agent 003 package get not_triaged'
+    output: ['due {"name":"test_deb_pkg","version":"1.0.0","architecture":"amd64","vendor":"Wazuh wazuh@wazuh.com","item_id":"1"}',
+             'ok {"status":"SUCCESS"}']
+    stage: "agent sys_programs getting package"
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/data/agent_messages.yaml [hunk 15]
```diff
@@ -172,62 +172,129 @@
...
+  -
+    input: 'agent 003 package get'
+    output: ['due {"name":"test_deb_pkg","version":"1.0.0","architecture":"amd64","vendor":"Wazuh wazuh@wazuh.com","item_id":"1"}',
+             'due {"name":"test_rpm_pkg","version":"1.0.0","architecture":"amd64","vendor":"Wazuh wazuh@wazuh.com","item_id":"1"}',
+             'ok {"status":"SUCCESS"}']
+    stage: "agent sys_programs getting package"
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/data/agent_messages.yaml]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 16]
```diff
@@ -76,6 +77,36 @@ def pre_insert_agents():
         assert data[0] == 'ok', f'Unable to update agent {id}'
 
 
+@pytest.fixture(scope="function")
+def pre_set_sync_info():
+    """Asign the last_attempt value to last_completion in sync_info table"""
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 8 above)
Comment (MiguelazoDS) [reply tests/integration/test_wazuh_db/test_wazuh_db.py]: Fixed

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**